server/src/com/cloud/network/lb/LoadBalancingRulesManagerImpl.java

```diff
@@ -2124,10 +2121,11 @@ public LoadBalancer updateLoadBalancerRule(UpdateLoadBalancerRuleCmd cmd) {
             throw new InvalidParameterValueException("Modifications in lb rule " + lbRuleId + " are not supported.");
         }
 
+        LoadBalancerVO tmplbVo = _lbDao.findById(lbRuleId);
         boolean success = _lbDao.update(lbRuleId, lb);
 
         // If algorithm is changed, have to reapply the lb config
-        if (algorithm != null) {
+        if ((algorithm!= null) && (tmplbVo.getAlgorithm().compareTo(algorithm)!=0)){
```

Comment:
It does the same work, but there is one more check to avoid the case when the existing algorithm is same as the new algorithm(No need to apply again which is redundant).